strax/plugin.py

```diff
@@ -608,6 +608,9 @@ def compute(self, **kwargs):
             loop_over = self.loop_over
         else:
             loop_over = self.deps[self.depends_on[0]].data_kind
+            if isinstance(loop_over, dict):
```

Comment:
You have to changes this line in `isinstance(loop_over, (dict, immutabledict))` since the DAQReader has an immutable dict as data_kind. See

```
plugins = st._get_plugins(targets=('events',), run_id='0')
plugins['records'].deps[plugins['records'].depends_on[0]].data_kind
```